fastestimator/op/tensorop/loss/mix_loss.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Any, Dict, List, TypeVar, Union
+
+import tensorflow as tf
+import torch
+
+import fastestimator as fe
+from fastestimator.backend import roll
+from fastestimator.op.tensorop.loss import LossOp
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor)
+
+
+class MixUpLoss(LossOp):
+    """This class should be used in conjunction with MixUpBatch and CutMixBatch to perform mix-up training, which helps to reduce
+    over-fitting, stabilize GAN training, and harden against adversarial attacks (https://arxiv.org/abs/1710.09412)
+    Args:
+        loss (func): A loss object which can be invoked like "loss(true, pred)". It's reduction method will be
+        overridden to 'none'
+        lam: The key of the lambda value generated by MixUpBatch
+    """
+    def __init__(self, loss: LossOp, lam: str, average_loss: bool = True):
+        self.loss = loss
+        self.loss.average_loss = False
+        if len(loss.outputs) == 1:
+            loss.outputs = loss.outputs[0]
```

Comment:
don't do this, it might cause problems elsewhere in the system where outputs are always expected to be in array format. use iteration in the forward method instead